external/curl/Makefile

```diff
@@ -0,0 +1,121 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+# curl/Makefile
+
+-include $(TOPDIR)/.config
+-include $(TOPDIR)/Make.defs
+
+ASRCS =
+# CURL common & feature specific source files
+CSRCS = amigaos.c curl_threads.c http.c nonblock.c socks_sspi.c asyn-ares.c \
+dict.c http_chunks.c nwlib.c speedcheck.c asyn-thread.c dotdot.c http_digest.c nwos.c \
+splay.c base64.c easy.c http_negotiate.c openldap.c ssh.c conncache.c escape.c http_ntlm.c \
+parsedate.c ssh-libssh.c connect.c file.c http_proxy.c pingpong.c strcase.c content_encoding.c \
+fileinfo.c idn_win32.c pipeline.c strdup.c cookie.c formdata.c if2ip.c pop3.c strerror.c \
+curl_addrinfo.c ftp.c imap.c progress.c strtok.c curl_des.c ftplistparser.c inet_ntop.c rand.c \
+strtoofft.c curl_endian.c getenv.c inet_pton.c rtsp.c curl_fnmatch.c getinfo.c krb5.c security.c \
+telnet.c curl_gethostname.c gopher.c select.c tftp.c curl_gssapi.c hash.c llist.c sendf.c timeval.c \
+curl_memrchr.c hmac.c md4.c setopt.c transfer.c curl_multibyte.c hostasyn.c md5.c sha256.c url.c \
+curl_ntlm_core.c hostcheck.c memdebug.c share.c version.c curl_ntlm_wb.c hostip4.c mime.c slist.c \
+warnless.c curl_path.c hostip6.c mprintf.c smb.c wildcard.c curl_rtmp.c	hostip.c multi.c smtp.c \
+x509asn1.c curl_sasl.c hostsyn.c netrc.c socks.c curl_sspi.c http2.c non-ascii.c socks_gssapi.c
+
+# Authentication/Crypto specific source files
+CSRCS+= vauth/cleartext.c vauth/cram.c vauth/digest.c vauth/digest_sspi.c vauth/krb5_gssapi.c \
+vauth/krb5_sspi.c vauth/ntlm.c vauth/ntlm_sspi.c vauth/oauth2.c vauth/spnego_gssapi.c \
+vauth/spnego_sspi.c vauth/vauth.c
+
+# TLS/SSL specific source files
+CSRCS+= vtls/vtls.c vtls/polarssl_threadlock.c
+ifeq ($(CONFIG_AXTLS),y)
+CSRCS+= vtls/axtls.c
+endif
+ifeq ($(CONFIG_CYASSL),y)
+CSRCS+= vtls/cyassl.c
+endif
+ifeq ($(CONFIG_DARWINSSL),y)
+CSRCS+= vtls/darwinssl.c
+endif
+ifeq ($(CONFIG_GNUTLS),y)
+CSRCS+= vtls/gtls.c
+endif
+ifeq ($(CONFIG_GSKIT),y)
+CSRCS+= vtls/gskit.c
+endif
+ifeq ($(CONFIG_MBEDTLS),y)
+CSRCS+= vtls/mbedtls.c
+endif
+ifeq ($(CONFIG_NSS),y)
+CSRCS+= vtls/nss.c
+endif
+ifeq ($(CONFIG_OPENSSL),y)
+CSRCS+= vtls/openssl.c
+endif
+ifeq ($(CONFIG_POLARSSL),y)
+CSRCS+= vtls/polarssl.c
+endif
+ifeq ($(CONFIG_SCHANNEL),y)
+CSRCS+= vtls/schannel.c
+endif
+
+AOBJS		= $(ASRCS:.S=$(OBJEXT))
+COBJS		= $(CSRCS:.c=$(OBJEXT))
+
+SRCS		= $(ASRCS) $(CSRCS)
+OBJS		= $(AOBJS) $(COBJS)
+
+# Make Own Archive for CURL
+BIN		= libcurl$(LIBEXT)
```

Comment:
Could you let me know the reason why you make it as separated libray from external?
All of others are merged into external library. Why is this special?

Reply:
1. Most of the modules in extern folder are a kind of mandatory for a basic Thing. Be it Wi-Fi, TLS and network utilities.
2. As you mentioned, CURL is not mandatory module. Basic HTTP 1.1 communication works with WebClient.
3. Cohesive CURL library similar to IoTivity (IoTivity has 6 cohesive libraries).
4. Quick memstat details specific to CURL, not mixed up with other extern modules.

Reply:
1. I don't agree. How about gRPC? AWS?
2. I don't understand what you want to say.
3. Iotivity and IoTjs have different build script, not Makefile. But libcurl has Makefile.
4. If we focus getting memstat quickly and easily, we should change almost module. But that is not good at outside developer. Samsung internal checking is not first priority on opensource.

Our basic rule (almost other opensource as well) is belonging to parent library like external. Libcxx can be splitted, but not libcurl I think.

Reply:
If it is Basic rule, we will comply to it.
I will make CURL as part of extern library.

Reply:
@bhargava-cs Thank you for understading.
If you think separated library is right, please let us know. But I think above is not enough. I will discuss it with my team and let you know.